### ISSUE TITLE
Constant high cpu load, probably caused by rewrites of invalid names (Knative & VirtualService)

### ISSUE THREAD

**Issue by TheEvilCoder42** [solo-io/gloo#1581]
**Describe the bug**

As I described in #920; Gloo pod constantly has a high cpu load, probably caused by rewrites of invalid names generated by Knative and VirtualService.

**To Reproduce**
1. Install Gloo (0.20.11) with Knative & Gateway
`glooctl install knative --install-knative=true --install-knative-version=0.9.0 --values gloo-config.yaml`

gloo-config.yaml:

```
discovery:
  enabled: false

gateway:
  enabled: true

gatewayProxies:
  gatewayProxyV2:
    service:
      extraAnnotations:
        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
        # Ensure the ELB idle timeout is less than nginx keep-alive timeout. By default,
        # NGINX keep-alive is set to 75s. If using WebSockets, the value will need to be
        # increased to '3600' to avoid any potential issues.
        service.beta.kubernetes.io/aws-load-balancer-connection-draining-enabled: "true"
        service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: "60"
        service.beta.kubernetes.io/aws-load-balancer-ssl-cert: {My cert}
        service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "https"
```
2. Spin up some example-apps (in different namespaces)
`kubectl apply -f tenant-1.yaml`

tenant-1.yaml:
```
apiVersion: v1
kind: Namespace
metadata:
  name: tenant-1
---
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: entry-route
  namespace: tenant-1
spec:
  virtualHost:
    domains:
      - 'tenant-1.foo.bar'
    routes:
      - matcher:
          regex: .*
        routeAction:
          single:
            kube:
              port: 80
              ref:
                name: knative-external-proxy
                namespace: gloo-system
        routePlugins:
          hostRewrite:
            hostRewrite: example-app.tenant-1.foo.bar
---
apiVersion: serving.knative.dev/v1alpha1
kind: Service
metadata:
  name: example-app
  namespace: tenant-1
spec:
  template:
    spec:
      containers:
        - image: {Some example image}
          env:
            - name: VERSION
              value: v1
            - name: NUM_WORKERS
              value: "4"
            - name: NAMESPACE
              value: "tenant-1"
```

3. Look at the gloo pod logs and constantly see the following:

```
{"level":"warn","ts":1572600310.3313935,"logger":"gloo.v1.event_loop.setup.v1.event_loop.envoyTranslatorSyncer.translator.compute_route_config.http-routes","caller":"utils/sanitize.go:19","msg":"illegal character(s) '.' in virtual host name [tenant-1.example-app-0] will be replaced by '_'"}
...
{"level":"warn","ts":1572600324.6679225,"logger":"gloo.v1.event_loop.setup.v1.event_loop.envoyTranslatorSyncer.translator.compute_route_config.listener-::-80-routes","caller":"utils/sanitize.go:19","msg":"illegal character(s) '.' in virtual host name [tenant-1.entry-route] will be replaced by '_'"}
...
{"level":"info","ts":1572600879.1981657,"logger":"gloo.v1.event_loop.setup.v1.event_loop.envoyTranslatorSyncer","caller":"syncer/envoy_translator_syncer.go:122","msg":"Setting xDS Snapshot","key":"gloo-system~knative-external-proxy","clusters":524,"listeners":1,"routes":1,"endpoints":524}
{"level":"info","ts":1572600879.4428105,"logger":"gloo.v1.event_loop.setup.v1.event_loop.envoyTranslatorSyncer","caller":"syncer/envoy_translator_syncer.go:122","msg":"Setting xDS Snapshot","key":"gloo-system~knative-internal-proxy","clusters":524,"listeners":0,"routes":0,"endpoints":524}
{"level":"info","ts":1572600879.4731371,"logger":"gloo.v1.event_loop.setup.v1.event_loop.envoyTranslatorSyncer","caller":"syncer/envoy_translator_syncer.go:135","msg":"end sync 15070998972998164608"}
```

**Expected behavior**

Gloo pod shouldn't have to rewrite invalid names, since this probably causes the unneeded high cpu load, resulting in wasted resources.

**Additional context**

Explaining the names in the logs:

There's a namespace `tenant-1` with a service `example-app` and a virtual service `entry-route`.

The knative service gets added as `tenant-1.example-app-0` and the virtual service as `tenant-1.entry-route`.

When I sequentially add 100 tenants, gloo pod has a high load (1700m) and constantly logs the mentioned (for each knative service & virtual service).

**Comment by marcogschmidt** [comment]
@TheEvilCoder42 We changed the log level for that message with [this PR](https://github.com/solo-io/gloo/pull/1605) and the change has been released with version [0.21.0](https://github.com/solo-io/gloo/releases/tag/v0.21.0). This should solve the high cpu load issue when running in large clusters, as the bulk was caused by the log I/O. 

I am leaving this open until we address the rest of the issue, namely Gloo writing choosing problematic names in the first palace (see #920).

**Comment by EItanya** [comment]
Similar to #1711, closing in favor of that
